packages/odyssey-react-mui/src/PasswordField.tsx

```diff
@@ -0,0 +1,175 @@
+  hint?: string;
+  /**
+   * The id of the `input` element.
+   */
+  id?: string;
+  /**
+   * If `true`, the component is disabled.
+   */
+  isDisabled?: boolean;
+  /**
+   * It prevents the user from changing the value of the field
+   */
+  isReadOnly?: boolean;
+  /**
+   * If `true`, the `input` element is required.
+   */
+  isRequired?: boolean;
+  /**
+   * The label for the `input` element.
+   */
+  label: string;
+  /**
+   * Callback fired when the `input` element loses focus.
+   */
+  onBlur?: FocusEventHandler<HTMLInputElement | HTMLTextAreaElement>;
+  /**
+   * Callback fired when the value is changed.
+   */
+  onChange?: ChangeEventHandler<HTMLTextAreaElement | HTMLInputElement>;
+  /**
+   * Callback fired when the `input` element get focus.
+   */
+  onFocus?: FocusEventHandler<HTMLInputElement | HTMLTextAreaElement>;
+  /**
+   * The short hint displayed in the `input` before the user enters a value.
+   */
+  placeholder?: string;
+  /**
+   * The value of the `input` element, required for a controlled component.
+   */
+  value?: string;
+};
+
+const PasswordField = forwardRef<HTMLInputElement, PasswordFieldProps>(
+  (
+    {
+      autoCompleteType,
+      autoFocus,
+      errorMessage,
+      hint,
+      id: idOverride,
+      isDisabled = false,
+      isReadOnly,
+      label,
+      onChange,
+      onFocus,
+      onBlur,
+      placeholder,
+      value,
+    },
+    ref
+  ) => {
+    const [inputType, setInputType] = useState("password");
+
+    const togglePasswordVisibility = useCallback(() => {
+      setInputType((inputType) =>
+        inputType === "password" ? "text" : "password"
+      );
+    }, []);
+
+    const renderFieldComponent = useCallback(
+      ({ ariaDescribedBy, id }) => (
+        <InputBase
+          aria-describedby={ariaDescribedBy}
+          autoComplete={autoCompleteType}
+          /* eslint-disable-next-line jsx-a11y/no-autofocus */
+          autoFocus={autoFocus}
+          endAdornment={
+            <InputAdornment position="end">
+              <IconButton
+                aria-label="toggle password visibility"
+                edge="end"
+                onClick={togglePasswordVisibility}
+              >
+                {inputType === "password" ? <EyeIcon /> : <EyeOffIcon />}
+              </IconButton>
+            </InputAdornment>
+          }
+          id={id}
+          onChange={onChange}
+          onFocus={onFocus}
+          onBlur={onBlur}
+          placeholder={placeholder}
+          readOnly={isReadOnly}
+          ref={ref}
+          type={inputType}
+          value={value}
+        />
+      ),
+      [
+        autoCompleteType,
+        autoFocus,
+        togglePasswordVisibility,
+        inputType,
+        onChange,
+        onFocus,
+        onBlur,
+        placeholder,
+        isReadOnly,
+        ref,
+        value,
+      ]
+    );
+
+    return (
+      <Field
+        errorMessage={errorMessage}
+        hasVisibleLabel={false}
```

Comment:
Why is this `false`?

Reply:
Whoops! fc187980da8983f59813617d5950291202fc5b7e